### ISSUE TITLE
Listener to files that get deleted

### ISSUE THREAD

**Issue by PopTudor** [danikula/AndroidVideoCache#49]
Hello ! I have a requirement to store in the database what song was cached and to delete from database the song that was deleted from cache. I need to keep account on what songs do I have in cache and what songs get deleted. Can I do this ?

**Comment by gouravd** [comment]
+1 for this enhancement/feature along with some sort of basic encryption

**Comment by stangabogdan** [comment]
I created a pull request with this feature added.

It's easy to integrate. When you crated your proxy builder you just add .setFileListener as in this example:

```
mProxy = new HttpProxyCacheServer.Builder(context)
               .setFileDeleteListener(new FileDeleteListener() {
                    @Override
                    public void onFileDeleted(String fileName) {
                         //fileName = the name of the just deleted file from cache
                    }
                })
               .build();
```

**Comment by danikula** [comment]
You can just check was file cached or not with help of `HttpProxyCacheServer.isCached(String url)`. This method is available in version 2.4.0. See [readme](https://github.com/danikula/AndroidVideoCache#listen-caching-progress)
